fix: mobile layout collision in header and services list (fixes #3294)

sarbojitrana opened the pull request:
## Title: fix: responsive layout breakdown on mobile screens (Hero & Services)


## Description
This PR addresses the responsive design issues reported in keploy/keploy#3294, where UI elements were colliding and overlapping on tablet and mobile viewports (500px - 900px).

## The Fixes:

Hero Section: The .home-social icons were using absolute positioning (right: 5.4rem), causing them to float on top of the content on smaller screens. I added a media query to switch them to a relative Flexbox layout on mobile.

Evaluation Criteria: The .services-list text had a negative top margin (-20px), which caused text to overlap with the numbering icons when the layout stacked vertically. I forced a margin reset in the mobile media query.

## Fixes keploy/keploy#3294

## Type of change
[x] Bug fix (non-breaking change which fixes an issue)


How Has This Been Tested?
I tested the layout changes locally using a Python HTTP server to ensure the DOM elements flow correctly without collision.

Test Environment: Arch Linux / Chrome & Firefox DevTools

Steps to Reproduce:

Run python3 -m http.server 8000 in the root directory.

Open http://localhost:8000.

Resize the browser viewport to 600px (Mobile width).

Verified that Social Icons now sit below the hero text.

Verified that "Evaluation Criteria" text sits below the number icons (1, 2, 3) without overlap.

## Screenshots :

**Before Fix**
<img width="790" height="970" alt="image" src="https://github.com/user-attachments/assets/6177da9a-dab4-4713-b80a-0294132628ae" />

<img width="790" height="970" alt="image" src="https://github.com/user-attachments/assets/1a13781d-7b22-40ba-88eb-5bb391825bb5" />


**After Fix**

<img width="790" height="970" alt="image" src="https://github.com/user-attachments/assets/5adea340-af8d-4a2e-8c8e-0b799e85322d" />
<img width="790" height="970" alt="image" src="https://github.com/user-attachments/assets/c59b4c83-29ea-4cdd-a705-65cf9c29562e" />


Checklist:
[x] My code follows the style guidelines of this project

[x] I have performed a self-review of my own code

[x] I have commented my code, particularly in hard-to-understand areas

[ ] I have made corresponding changes to the documentation

[x] My changes generate no new warnings

[ ] I have added tests that prove my fix is effective or that my feature works

[ ] New and existing unit tests pass locally with my changes

[ ] Any dependent changes have been merged and published in downstream modules

[x] I have checked my code and corrected any misspellings

[ ] I have tagged the reviewers in a comment below incase my pull request is ready for a review

[x] I have signed the commit message to agree to Developer Certificate of Origin (DCO) (to certify that you wrote or otherwise have the right to submit your contribution to the project.) by adding "--signoff" to my git commit command.

sarbojitrana committed:
fix: mobile layout collision in header and services list (fixes #3294)

Signed-off-by: sarbojitrana <sarbojitrana47c@gmail.com>

sarbojitrana commented:
Hi Team, 

I've submitted the fix for issue #3294. 

It looks like the `CLA Assistant` workflow crashed immediately (failed after 5s) with an internal error, so it never posted the link for me to sign the CLA. 

Could you please review the PR and manually handle the CLA check? The DCO check has passed.

amaan-bhati reviewed (changes requested):
Hey @sarbojitrana Thanks for raising this pr, can you please center align these two buttons and ensure that they are visible in all screen sizes?

<img width="537" height="467" alt="Image" src="https://github.com/user-attachments/assets/cb8caa98-7b07-4511-9f48-d3f511213b96" />

sarbojitrana commented:
@amaan-bhati 
sure i'll work on it

sarbojitrana committed:
fix: center align buttons and prevent horizontal scroll on mobile

Signed-off-by: sarbojitrana <sarbojitrana47c@gmail.com>

sarbojitrana committed:
fix: correct indentation in workflow file

Signed-off-by: sarbojitrana <sarbojitrana47c@gmail.com>

sarbojitrana commented:
<img width="992" height="982" alt="image" src="https://github.com/user-attachments/assets/00dd90ab-85e8-4f76-892c-ab45b75930c8" />
<img width="992" height="982" alt="image" src="https://github.com/user-attachments/assets/804044fc-d35b-4988-94cc-3b332d689932" />
Hey @amaan-bhati, I've updated the CSS to center the buttons and prevent horizontal scrolling on mobile. I also fixed the indentation error in the build_and_check workflow, so the CI build is passing now.

The CLA check is still failing, but it looks like an internal server error on the bot's side. Ready for review!

amaan-bhati reviewed (changes requested):
Hey @sarbojitrana Can you please take a look at the following?
- idts the header is supposed to be like that on smaller screen sizes, it only pops up when the hamburger is clicked
<img width="191" height="45" alt="Image" src="https://github.com/user-attachments/assets/7986d772-dbeb-4df2-b213-c62ee58a2043" />
</br>
- the heading at the hero section gets lost above the bg image, can you please make it a little bolder or maybe add some low opacity bg that makes it easier to locate the description above the bg image. Also, the two buttons are still not visible on smaller screen sizes, kindly test it on 3-4 devices on the inspect tab to ensure that the buttons are visible </br>
<img width="202" height="263" alt="Image" src="https://github.com/user-attachments/assets/02187262-3809-44c5-9e77-3c30ad8e8080" /> </br>
- there are issues with the padding inside of a few buttons
</br>
<img width="134" height="80" alt="Image" src="https://github.com/user-attachments/assets/2db47046-1464-4e56-b867-89dd9860d22f" /> </br>
- elements overlapping </br>

<img width="150" height="298" alt="Image" src="https://github.com/user-attachments/assets/574fe792-bfa4-4439-beb1-598c76d776a3" />

<img width="141" height="302" alt="Image" src="https://github.com/user-attachments/assets/86782fce-1259-4826-8555-78816e22b747" />

sarbojitrana committed:
fix: align social icons vertically, fix text overlap, and restore image vibrancy

Signed-off-by: sarbojitrana <sarbojitrana47c@gmail.com>

sarbojitrana commented:
Hey @amaan-bhati  thanks for the detailed feedback! I've pushed the fixes and tested on multiple device's resolutions. 
<img width="613" height="951" alt="image" src="https://github.com/user-attachments/assets/dc8556d8-e169-418f-85fa-6a1003ba067a" />
<img width="613" height="951" alt="image" src="https://github.com/user-attachments/assets/b97399e3-7e30-426a-b4ce-09d2d0053d6d" />
<img width="613" height="951" alt="image" src="https://github.com/user-attachments/assets/33dbf969-e8f1-4394-8bbb-e074cc212478" />
<img width="613" height="951" alt="image" src="https://github.com/user-attachments/assets/483a880b-0775-4213-9d3a-6fac843396f8" />
<img width="613" height="951" alt="image" src="https://github.com/user-attachments/assets/43b884b8-4cba-49ba-b147-f97acf059c52" />
<img width="613" height="951" alt="image" src="https://github.com/user-attachments/assets/a4e11443-1cd0-4b71-97f6-21f6372cbfaf" />
Please let me know if there is anything else that needs to be looked on.

amaan-bhati reviewed (changes requested):
Hey @sarbojitrana Thanks for the updates, most of it looks good to me now. I think only a bunch of finishing touch is left and then we are good to merge. Can you please take a looka at the following:
- the heading is partially underligned, i think the underline should be removed from the 'writers-program' written, also the image behind gets cut from the top. 
</br>

<img width="254" height="301" alt="Image" src="https://github.com/user-attachments/assets/5c435a30-632d-4125-a467-9751fa410ce6" />
</br>

- the social dock next to the menu hamburger is not reallly required, we can remove it because the socials are visible on the hamburger drop down anyways
</br>
<img width="110" height="352" alt="Image" src="https://github.com/user-attachments/assets/3115b423-0638-45bc-a68c-f0e2a26e252b" />
</br>
- `menu` written next to the hamburger is not really required since the hamburger it self is explanatory as the menu button
</br>
<img width="95" height="51" alt="Image" src="https://github.com/user-attachments/assets/d272ee38-6a00-4687-ba6a-0e4628a1aa0d" />
</br>
- the footer section is not centrally alligned with respect to other sections and the buttons
</br>

<img width="296" height="480" alt="Image" src="https://github.com/user-attachments/assets/19c703cc-9307-487a-a23a-a1035e3894bd" />

</br>

most of the things that were breaking are fixed now, we are goofd merge after these changes, good work.

sarbojitrana committed:
fix: finalize mobile layout alignment, visibility, and spacing per review

Signed-off-by: sarbojitrana <sarbojitrana47c@gmail.com>

sarbojitrana commented:
Hi @amaan-bhati , i've fixed the suggestions you gave in the review. please check it out

<img width="623" height="846" alt="image" src="https://github.com/user-attachments/assets/0377a281-6d32-49ab-81fd-ea3189148715" />
<img width="623" height="846" alt="image" src="https://github.com/user-attachments/assets/a5f1e72f-f499-42d0-a8d5-9191155419b5" />

amaan-bhati reviewed (changes requested):
Hey @sarbojitrana Thanks for the update, most of the things are fixed now, you have covered every requested change now. The only edge case we should be careful about is the hero section button in smaller devices like the iphone se, the button is getting cut from the buttom in case of that particular device, it might happen for the devices with similar dimension, i have attached a refernce screenshot for the same:

<img width="537" height="676" alt="Image" src="https://github.com/user-attachments/assets/05df4f25-1563-4a79-b11d-8e7c3f128262" />

sarbojitrana committed:
fix: fix the buttons in the hero sections for smaller devices

sarbojitrana committed:
fix: fix the buttons and scroll down link

sarbojitrana commented:
fixed it and tested for multiple screens

<img width="520" height="802" alt="image" src="https://github.com/user-attachments/assets/27e83096-a12d-4a55-9cb5-490709b9ff02" />


<img width="537" height="976" alt="image" src="https://github.com/user-attachments/assets/d263934d-b9de-43d9-b32e-b2a389a954cc" />

amaan-bhati reviewed (approved):
hey @sarbojitrana thanks for the update, everything looks fine to me, great that we covered most of the responsiveness issues, everything looks good to me⚡. thanks for contributing, kudos to you.🚀
